src/layers/models/layersManager.ts

```diff
@@ -106,28 +144,49 @@ class LayersManager {
     this.logger.info(`Successfully updated mosaic: '${mosaicName}'`);
   }
 
+  public getAllLinkedCaches(baseCacheNames: string[], mapproxyConfiguration: IMapProxyJsonDocument): string[] {
+    const linkedCaches: string[] = [];
+    const baseCacheNamesDuplicate: string[] = [...baseCacheNames];
+    baseCacheNamesDuplicate.forEach((currentCache) => {
+      if (mapproxyConfiguration.caches[currentCache]) {
+        if (currentCache.endsWith('-source')) {
+          if (linkedCaches.indexOf(currentCache) == -1) {
```

Comment:
Why don't you use other methods such as incldes and find? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find